3478/1/top_gems/terminal_output.rb

```diff
@@ -0,0 +1,35 @@
+require 'terminal-table'
+
+class TerminalOutput
+  HEADER = %w[Name Used_by Watchers Stars Forks Contributors Issues].freeze
+
+  def initialize(gem_top, top_size, contains_in_name)
+    @gem_top = gem_top.sort_by { |gem| gem[:total_score] }
+    @contains_in_name = contains_in_name
+    top_size ||= @gem_top.size
+    @top_size = top_size.to_i
+    @rows = []
+  end
+
+  def print_top
+    @gem_top.reverse.each do |gem|
+      next unless appropriate_name?(gem)
+      break if @rows.size >= @top_size
```

Comment:
try to avoid using `break` command. It's not really needed in 95% of cases. You could use `take(top_size)` in this case